cli/internal/cache/cache_fs_test.go

```diff
@@ -60,87 +31,81 @@ func TestPut(t *testing.T) {
 	//
 	// <dst>/the-hash/<src>/...
 
-	src := subdirForTest(t)
-	childDir := filepath.Join(src, "child")
-	err := os.Mkdir(childDir, os.ModeDir|0777)
+	src := turbopath.AbsoluteSystemPath(t.TempDir())
```

Comment:
Not blocking, but I'm really not a fan of casting for getting an instance of a path type:
 - it's hard to grep / find-usages for
 - they can be auto-inserted by the IDE
 - they don't communicate anything about why this is ok. 
 I would expect this to be something like `AbsoluteSystemPathFromUpstream`, which communicates that the author has verified that wherever the string is coming from produces absolute paths.

Reply:
The "rule" I've been following is that I've only been doing type casts in test code, and only on initialization from "raw" values (string primitives, or calls into stdlib). You'll see I didn't do casts in any application code or even in any of the more-hairy test code.

I don't feel strongly either way, so I'll defer to your opinion on this and get the casts swapped out before landing Monday.

Reply:
If we're going to make this a rule we should write a custom linter that disallows all casts into `turbopath` that aren't from a string:
<img width="522" alt="Screen Shot 2022-09-26 at 1 46 57 PM" src="https://user-images.githubusercontent.com/20542/192202319-44b59a4a-c714-45cc-8a23-412a6feda74d.png">

(Splitting this change into a separate PR.)

Reply:
Followup PR: https://github.com/vercel/turborepo/pull/2077